src/ConnectAuth.ts

```diff
@@ -33,53 +42,88 @@ export interface UpdateInstallationResponse<E> {
 
 export type VerifyInstallationResponse<E> = NewInstallationResponse | UpdateInstallationResponse<E>;
 
-export type VerifyRequestArgs<E> = CommonVerifyArgs<E, QueryStringHashType>;
-
-export interface VerifyRequestResponse<E> {
-  connectJwt: ConnectJwt;
-  storedEntity: E;
-}
-
 /**
  * Verifies a Connect request installation.
  * Use this function to make sure the request is valid before persisting any data.
  * This function handles both new installations and re-installations or installation updates.
  */
 export async function verifyInstallation<E>({
-  baseUrl,
+  asymmetricKeyProvider,
   authDataProvider,
+  authorizationMethod = 'any',
+  baseUrl,
   credentialsLoader,
   queryStringHashType,
 }: VerifyInstallationArgs<E>): Promise<VerifyInstallationResponse<E>> {
   const clientKey = authDataProvider.extractClientKey();
-
-  // Check issuer
   const rawConnectJwt = authDataProvider.extractConnectJwt();
+
+  // Parse unverified JWT
   let unverifiedConnectJwt;
   if (rawConnectJwt) {
     unverifiedConnectJwt = decodeUnverifiedConnectJwt(rawConnectJwt);
-    if (unverifiedConnectJwt.iss !== clientKey) {
-      throw new AuthError('Wrong issuer', {
-        code: AuthErrorCode.WRONG_ISSUER,
-        unverifiedConnectJwt,
-      });
+  }
+
+  // Check for a signed installation
+  if (
+    authorizationMethod === 'publicKey' ||
+    (authorizationMethod === 'any' && isAsymmetricAlgorithm(unverifiedConnectJwt?.alg))
+  ) {
+    if (!asymmetricKeyProvider) {
+      throw new Error('Missing asymmetricKeyProvider instance');
```

Comment:
Since the key provider was made option, I have to check whether it was provided or not.
I'm throwing a generic error because this is not an Auth error. This is more like an assertion that the developer consuming this library will remember to provide what is required